src/device/device.c

```diff
@@ -99,10 +99,6 @@ void init_device() {
   IFDEF(CONFIG_HAS_DISK, init_disk());
   IFDEF(CONFIG_HAS_SDCARD, init_sdcard());
   IFDEF(CONFIG_HAS_FLASH, init_flash());
-#ifndef CONFIG_SHARE
-  IFDEF(CONFIG_HAS_FLASH, load_flash_contents(CONFIG_FLASH_IMG_PATH));
-#endif // CONFIG_SHARE
```

Comment:
Maybe we should seperate these flash.c, device.c changes into another PR for review.

This PR still seems too large for us to understand what's going on here

Reply:
ok, i'll split this pr